packages/next-on-pages/src/buildApplication/buildSummary.ts

```diff
@@ -0,0 +1,179 @@
+ * @returns A list of items to be used in the build summary.
+ */
+function processItemsMap(items: Map<string, unknown>): string[] {
+	return [...items.keys()]
+		.filter(path => !/\.rsc$/.test(path))
+		.map(path => addLeadingSlash(stripIndexRoute(path.replace(/\.html$/, ''))))
+		.sort((a, b) => a.localeCompare(b));
+}
+
+/**
+ * Returns a prefix to be used in the printed build summary for an item's index.
+ *
+ * @param total Total number of items.
+ * @param idx Current item index.
+ * @returns The prefix for the log message.
+ */
+function getItemPrefix(total: number, idx: number): string {
+	if (total === 1) return '-';
+	if (idx === 0) return '┌';
+	if (idx === total - 1) return '└';
+	return '├';
+}
+
+/**
+ * Adds a group of items to the build summary, limiting the number of items to be displayed if
+ * necessary.
+ *
+ * Formats the items with stylised prefixes and indentations.
+ *
+ * @param summary Current summary.
+ * @param name The name of the group of items.
+ * @param rawItems List of items to be added to the summary.
+ * @param limit Maximum number of items to be displayed.
+ * @returns The updated summary.
+ */
+function addToSummary(
+	summary: string,
+	name: string,
+	rawItems: string[],
+	limit?: number
+): string {
+	if (rawItems.length === 0) return summary;
+
+	const items =
+		limit && rawItems.length > limit
+			? [
+					...rawItems.slice(0, limit - 1),
+					`... ${rawItems.length - limit - 1} more`,
+			  ]
+			: rawItems;
+
+	const formattedItems = items
+		.map((path, idx) => `  ${getItemPrefix(items.length, idx)} ${path}`)
+		.join('\n');
+
+	return `${summary}
+	
+${name} (${rawItems.length})
+${formattedItems}`;
+}
+
+/**
+ * Prints a build summary to the console.
+ *
+ * @param staticAssets List of static assets collected during the build.
+ * @param processedVercelOutput Results of processing the Vercel output directory.
+ * @param directoryProcessingResults Results of processing the output directory.
+ */
+export function printBuildSummary(
+	staticAssets: string[],
+	{ vercelOutput }: ProcessedVercelOutput,
+	{
+		functionsMap = new Map(),
+		prerenderedRoutes = new Map(),
+		wasmIdentifiers = new Map(),
+	}: Partial<DirectoryProcessingResults> = {}
+): void {
+	const middlewareFunctions = [...vercelOutput.entries()]
+		.filter(([, { type }]) => type === 'middleware')
+		.map(([path]) => path)
+		.sort((a, b) => a.localeCompare(b));
+	const routeFunctions = new Set(
+		processItemsMap(functionsMap).filter(
+			path => !middlewareFunctions.includes(path.replace(/^\//, ''))
+		)
+	);
+	const prerendered = processItemsMap(prerenderedRoutes);
+	let otherStatic = staticAssets
+		.filter(path => !prerenderedRoutes.has(path))
+		.sort((a, b) => a.localeCompare(b));
+
+	// Move the `/_next` and `/__next` assets to the end of the static assets list
+	otherStatic = [
+		...otherStatic.filter(path => !/^\/__?next/.test(path)),
+		...otherStatic.filter(path => /^\/__?next/.test(path)),
+	];
```

Comment:
What about doing another `sort`? 🙂 
```suggestion
	const otherStatic = staticAssets
		.filter(path => !prerenderedRoutes.has(path))
		.sort((a, b) => a.localeCompare(b))
		.sort((a, b) => {
			// Move the `/_next` and `/__next` assets to the end of the static assets list
			const aIsNextAsset = /^\/__?next/.test(a);
			const bIsNextAsset = /^\/__?next/.test(b);
			if (!aIsNextAsset && bIsNextAsset) return -1;
			if (aIsNextAsset && !bIsNextAsset) return 1;
			return 0;
		});
```

____

or even integrating the logic in the existing `sort` if you prefer:
```suggestion
	const otherStatic = staticAssets
		.filter(path => !prerenderedRoutes.has(path))
		.sort((a, b) => {
			// Note: `/_next` and `/__next` assets are sorted to the end of the static assets list
			const aIsNextAsset = /^\/__?next/.test(a);
			const bIsNextAsset = /^\/__?next/.test(b);
			if (!aIsNextAsset && bIsNextAsset) return -1;
			if (aIsNextAsset && !bIsNextAsset) return 1;

			return a.localeCompare(b);
		});
```